### ISSUE TITLE
Suggestion: Exclude command

### ISSUE THREAD

**Issue by MotikaCraft-TheJoshua** [kangarko/ChatControl#2497]
### Summary

New moderation command for excluding: `/chc exclude [username]`
When we execute `/chc exclude PlayerABC`, PlayerABC will from now on be excluded from chat in a way where only HE will be able to see his messages being sent to chat, simply put, exclude command will work just like if every player ever (even the ones who didn't ever join but will join in the future) were to execute `/ignore PlayerABC` and they will from now on not be able to see any PlayerABC's messages, but PlayerABC will be able to see both from player who ignored them and their own.

TL;DR:
1. Admin executes `/chc exclude PlayerABC`
2. PlayerABC will be able to chat and see their own messages just like before, but nobody else ever will be able to see their messages; making like everyone is ignoring PlayerABC.

### What would happen if we didn't implement this feature? Why not having this feature is a problem?

I think this feature is not present in any chat moderation plugin literally, so it would be really beneficial here.

**Comment by kangarko** [comment]
Interesting, maybe this could be an addon or even a separate moderation plugin

**Comment by MotikaCraft-TheJoshua** [comment]
I donćt think this way

I already thought of this to recommend to some moderation/punishments plugins but i did not because i think chat management plugin itself should control their own system

I don't trust (for example) LiteBans controlling my chat that is managed by CHCRED, i know that probably any case is possible, i just would feel more safe if this is part of my chat management plugin.

> Interesting, maybe this could be an addon or even a separate moderation plugin

Thank you for your response!

**Comment by MotikaCraft-TheJoshua** [comment]
I think this should be there in some specific cases when you ban 1 player over and over again and they come over when you (admin) are not online, spam their advertisements... then you literally do `/chc exclude Username` and they will think they're advertising but actually:
- nobody see their messages except them so they think everyone see it because nothing is noticing them about it
- nobody see their private messages except they will see that it is sent, but receiver will actually receive nothing

I strongly believe this should not be in separate from CHCRED, thank you so much in advance for further discussion.

**Comment by kangarko** [comment]
Thanks for your feedback, I consider it however my concern is that our codebase is getting clunkier with every feature, plus there is a learning curve of the plugin for beginners which I dont like. I like simplicity and the original vision of chat management, not all-in-one plugin that is failing on its own weight

It's a good feature what you suggested, perhaps I can make open source addon and submit to builtbybit for a small price, people can choose to support me or compile the source themselves and it would hook into chatcontrol directly

**Comment by MotikaCraft-TheJoshua** [comment]
> Thanks for your feedback, I consider it however my concern is that our codebase is getting clunkier with every feature, plus there is a learning curve of the plugin for beginners which I dont like. I like simplicity and the original vision of chat management, not all-in-one plugin that is failing on its own weight
> 
> It's a good feature what you suggested, perhaps I can make open source addon and submit to builtbybit for a small price, people can choose to support me or compile the source themselves and it would hook into chatcontrol directly

i second this, could be an addon for moderation specifically designed to work with CHCRED (somewhat similar system like CombatLogX uses(?) for addons maybe) and to have more moderation features like the one i suggested.

**Comment by kangarko** [comment]
Sure. No ETA tho. Keeping this open for implementation.

**Comment by nezoxclipss** [comment]
Also do not use CombatLogX

**Comment by kangarko** [comment]
Hey! I believe this was added in the meanwhile in the form of the "then deny silently" operator. 

It prevents the message from appearing for others but sends it to sender, making him believe it was really sent out. Useful for blocking ads so that the player will not continue to spam them but will believe his advertisement has been seen (where in reality only admins have seen it and can take proper massive action).

I believe you can combine this with "save key" and related operators as demonstrated in rules/chat.rs but instead, you'd be creating a new rule in rules/commands.rs for the exclude command and toggle the ignore status. The make a catch-all rule "match .*" and check for the saved key and ignore the message silently.

Example of the save key being used: https://i.imgur.com/frbvOtZ.png
